src/Core/Application/Identity/Roles/IRoleService.cs

```diff
@@ -4,7 +4,7 @@ public interface IRoleService : ITransientService
 {
     Task<List<RoleDto>> GetListAsync();
 
-    Task<List<PermissionDto>> GetPermissionsAsync(string id, CancellationToken cancellationToken);
+    Task<List<PermissionDto>> GetPermissionsAsync(string id, CancellationToken cancellationToken, bool retrieveOnlyFromRole = true);
```

Comment:
This doesn't make a lot of sense... if the bool is false, then the id is not necessary... then it also doesn't have anything to do with roles, but just with permissions...

I would make a new api for this... GET /api/permissions or something like that...

Reply:
Hi, maybe I didnt explained right. 
The RoleId is always necessary, in case the new param "retrieveOnlyFromRole " is false, it retrieves all permissions, but the permissions for the roleId contains the property "Enabled" = true. 
I Added this so in the Client, we can list all the permissions(even the ones that role doesnt have) and to avoid make two calls to the server. 
I Used this method because it was already created as the update.

Reply:
Oh ok... I understand now. I misinterpreted the meaning of the parameter... 

Just thinking out loud here... I understand that you want to limit the number of calls... but still... I think it would be better to actually include the permissions into the GetRole call, so you only need to do one call there on the client. The full list of roles is already on the client in the shared project... so no need to get these from the server?

Reply:
I've mentioned the subject in a new issue #381

Reply:
Ok,Maybe is better to use the ones in Client, the ones on Client, are manually updated ou autogenerated?

Reply:
They are manually updated for now... still need to find a way to actually share the project...

Reply:
Is there a way to include them in the generate code?

Reply:
Nope... the generated code is only for the ApiClients. The rest of the shared library (like permissions) don't really have anything to do with the ApiClients.

We probably could create a script that simply copies over the files... Anyways, I'll create another issue for this ;-)

Reply:
See https://github.com/fullstackhero/blazor-wasm-boilerplate/issues/87